src/views/open-api-docs-view-v2/server.ts

```diff
@@ -3,138 +3,314 @@
+	 * is the latest stable version, or if there are no stable versions,
+	 * then the latest version regardless of release stage.
+	 */
+	// Fetch version data
+	const rawVersionData = Array.isArray(schemaSource)
+		? schemaSource
+		: await fetchCloudApiVersionData(schemaSource)
+	const versionData = transformVersionData(rawVersionData)
+	// Determine the default version
+	const defaultVersion = findDefaultVersion(versionData)
+	// Parse the default version
+	const schemaFileString =
+		typeof defaultVersion.sourceFile === 'string'
+			? defaultVersion.sourceFile
+			: await fetchGithubFile(defaultVersion.sourceFile)
+	const schemaData = await parseAndValidateOpenApiSchema(
+		schemaFileString,
+		schemaTransforms
+	)
+	// Extract operation objects, and map to slugs
+	const operationObjects = getOperationObjects(schemaData)
+	const operationSlugs = operationObjects.map((operation) => {
+		return slugifyOperationId(operation.operationId)
+	})
+	// Generate path objects for each operation slug
+	const pathObjects = operationSlugs.map((operationSlug) => {
+		return { params: { page: [operationSlug] } }
+	})
+	// Return paths
+	return {
+		paths: [{ params: { page: [] } }, ...pathObjects],
+		fallback: 'blocking',
+	}
+}
+
+/**
+ * Wrapper around `generateStaticProps` that handles the common production
+ * use case of fetching version data from GitHub.
+ */
+export async function generateStaticPropsVersioned(
+	pageConfig: OpenApiDocsViewV2Config,
+	params: string[] | never
+): Promise<{ props: OpenApiDocsViewV2Props } | { notFound: true }> {
+	// Fetch version data
+	const rawVersionData = Array.isArray(pageConfig.schemaSource)
+		? pageConfig.schemaSource
+		: await fetchCloudApiVersionData(pageConfig.schemaSource)
+	const versionData =
+		typeof pageConfig.transformVersionData === 'function'
+			? pageConfig.transformVersionData(rawVersionData)
+			: rawVersionData
+	// Parse the URL context, to determine the version and operationSlug.
+	const urlContext = parseOpenApiV2UrlContext(params)
+	// Return static props, or may return `{ notFound: true }`.
+	return await generateStaticProps({
+		...pageConfig,
+		versionData,
+		urlContext,
+	})
+}
 
 /**
  * Build static props for an OpenAPI docs view.
  *
  * There are two main views:
  * - Landing view, for the basePath, when no operationSlug is provided
  * - Operation view, for the specific operationSlug that's been provided
+ *
+ * This function expects `versionData`, an array of objects. This accommodates
+ * both the HCP-centric use case, where we fetch `versionData` from GitHub,
+ * with each `versionData` entry referencing a `GithubFile` as its `sourceFile`,
+ * as well as the more general use case, where `versionData` can be an array
+ * with a single object, with the entry passing the schema file string
+ * directly as the `sourceFile`.
  */
-export async function getStaticProps({
+export async function generateStaticProps({
```

Comment:
Third, we modify the interface of `generateStaticProps`, to support versioning. We also shift some logic around, including figuring out the `targetOperation` in this function body (rather than in `getOperationContentProps` as we did previously), and adding page titles via the `metadata` return value.